contracts/core/OriginCore.sol

```diff
@@ -285,7 +302,52 @@ contract OriginCore is OriginCoreInterface, OriginCoreConfig {
         external
         returns (bool success_)
     {
-        revert("Method not implemented.");
+        /**
+         * This check if valid meta-block proposal exists for given
+         * kernel hash and transition hash. This will revert transaction
+         * if proposal doesn't exist.
+         */
+        checkMetaBlockProposal(
+            _kernelHash,
+            _transition
+        );
```

Comment:
Maybe rather:
```solidity
require(
    existsMetaBlockProposal(_kernelHash, _transition),
    "A vote can only be verified for an existing meta-block proposal."
);
```

"checkMetaBlockProposal" is a little ambiguous. That's also why you had to add the comment above. You don't need that comment if you change it to something like what I proposed. Then the code is self-explanatory. Just an idea.